src/bmp.imageio/bmpinput.cpp

```diff
@@ -38,8 +38,8 @@ OIIO_PLUGIN_EXPORTS_BEGIN
 
     OIIO_EXPORT int bmp_imageio_version = OIIO_PLUGIN_VERSION;
     OIIO_EXPORT const char* bmp_imageio_library_version () { return NULL; }
-    OIIO_EXPORT ImageInput *bmp_input_imageio_create () {
-        return new BmpInput;
+    OIIO_EXPORT ImageInput::unique_ptr bmp_input_imageio_create () {
```

Comment:
Just wondering if instead, we should keep on returning a plain `ImageInput` and wrap into a unique_ptr inside OIIO (possibly pointing the destructor to a `imageio_destroy` function pointer inside the image plugin.

Reply:
Yeah, we could do that. It means that every plugin must supply a new function that returns the local destroy function. You think that would be better?

Reply:
Yeah I think that would be clearer/safer.